pkg/kubelet/kubelet_pods.go

```diff
@@ -1469,6 +1469,19 @@ func (kl *Kubelet) generateAPIPodStatus(pod *v1.Pod, podStatus *kubecontainer.Po
 		}
 	}
 
+	// Perform a three-way merge between the statuses from the status manager,
+	// runtime, and generated status. If the pod is terminated, the phase MUST be
+	// terminal. Default to Failed in the case we can't find a non-running Phase.
+	if isTerminated && s.Phase != v1.PodFailed && s.Phase != v1.PodSucceeded {
+		if oldPodStatus.Phase == v1.PodFailed || oldPodStatus.Phase == v1.PodSucceeded {
+			s.Phase = oldPodStatus.Phase
+		} else if pod.Status.Phase == v1.PodFailed || pod.Status.Phase == v1.PodSucceeded {
+			s.Phase = pod.Status.Phase
+		} else {
+			s.Phase = v1.PodFailed
```

Comment:
Hrm... I don't think we should ever get into this chunk.  Why is this here?

Reply:
Is it possible for everything to say Running but we've entered syncTerminatedPod? Trying to be defensive

Reply:
It's possible we don't have all pod status yet, so all containers will eventually be reported as successful, we just haven't calculated it yet.  If you calculate too early, it'll mark those pods as failed.